irohad/ametsuchi/impl/postgres_command_executor.cpp

```diff
@@ -28,6 +29,42 @@
 #include "interfaces/common_objects/types.hpp"
 
 namespace {
+  struct PreparedStatement {
+    std::string command_name;
+    std::string command_base;
+    std::vector<std::string> permission_checks;
+  };
+
+  std::pair<std::string, std::string> compileStatement(
+      const PreparedStatement &statement) {
+    auto initial = boost::format(statement.command_base);
+
+    auto with_validation =
```

Comment:
Would be nice to comment most of the actions in that function, since it very generic, thus it can be easily misused